### DISCUSSION TITLE
Are strategies executed only once for all addresses or batch executed for chunks of addresses?

### DISCUSSION THREAD

**Discussion by glmaljkovich** [snapshot-labs/snapshot#923] [Q&A] [answered]
I'm a strategy developer. Say that I have an upcoming proposal where 2000 users are going to vote. Should I develop my strategy to fetch results for all 2000 users, or do you split the strategy calls on batches of say 500 addresses at a time?

**Comment by bonustrack** [comment, marked as answer]
@glmaljkovich Unless the contract has some heavy logic it should work for 2k users. If it fail you can leverage the strategy called "pagination" that will batch calls for X address at a time for any strategy.
